src/bootstrap/builder.rs

```diff
@@ -1146,6 +1150,7 @@ impl<'a> Builder<'a> {
             if mode == Mode::Std && compiler.stage == 0 && self.config.incremental {
                 rustflags.arg("-Aunused-attributes");
             }
+            rustdocflags.arg("-Dinvalid_codeblock_attributes");
```

Comment:
`invalid_codeblock_attributes` is a warn by default lint so there's no reason to pass this. The `-Dwarnings` above already makes it deny.

Reply:
No, above is `rustflags` and we're adding it to `rustdocflags`. Or are you referring to another than the one 10 lines above?

Reply:
Yeah, I was mistaken. We should really just start adding `-Dwarnings` to `rustdocflags` as well as `rustflags`.

Reply:
That would be problematic because of the broken links for the moment.

Reply:
This should only be `-D` if `deny_warnings` is on.

Reply:
No because we want this check all the time.

Reply:
...No? I don't see how that follows. It should be a warning only if the developer has not requested to deny warnings.

Reply:
This is all this PR is about: prevent malformed doc codeblocks attributes in the libstd and the error codes. If I let it as a warning, it'll never be fixed and the situation will remain the same.

Reply:
> That would be problematic because of the broken links for the moment.

What broken links? There shouldn't be any and passing `-Dwarnings` will prevent new ones being added. I can understand not wanting to pass `-Dwarnings` for the compiler docs so if that's the concern then just denying `invalid_codeblock_attributes` for now is fine. We should definitely pass `-Dwarnings` when building at least the `std` docs at some point though.

Also setting `rustdocflags` here will override the setting from https://github.com/rust-lang/rust/blob/4cd0ee9343da86d9770bf0a514a682d240e0dce8/src/bootstrap/doc.rs#L530-L534 and https://github.com/rust-lang/rust/blob/4cd0ee9343da86d9770bf0a514a682d240e0dce8/src/bootstrap/doc.rs#L627

Those need to be changed to use the same mechanism for setting `rustdocflags` as is used here.

Reply:
> What broken links?

The intra-doc ones. We have a few warnings when generating (std) docs. But yes, once intra-doc links are stabilized, I intend to change it to `-Dwarnings`.

I'll update the `rustdocflags`.

Reply:
> This is all this PR is about: prevent malformed doc codeblocks attributes in the libstd and the error codes. If I let it as a warning, it'll never be fixed and the situation will remain the same.

This is confusing to me as an assertion. On CI, deny warnings is always enabled. Locally, when iterating, it can be useful to have warnings as just warnings. I imagine that applies to this one too.

Reply:
Oh, this is an important information. Then I guess it's acceptable.